internal/rbd/driver/driver.go

```diff
@@ -224,7 +224,9 @@ func (r *Driver) setupCSIAddonsServer(conf *util.Config) error {
 	}
 
 	if conf.IsNodeServer {
-		rs := casrbd.NewReclaimSpaceNodeServer()
+		rs := casrbd.NewReclaimSpaceNodeServer(&rbd.NodeServer{
+			VolumeLocks: util.NewVolumeLocks(),
```

Comment:
same here as well can you please check?

Reply:
you might need to pass `NewNodeServer(r.cd)`

Reply:
`NewNodeServer()` needs other params to be passed. So instead created NodeServer struct with volumeLocks
https://github.com/ceph/ceph-csi/blob/9574fa881bc6e2840ebb135d49fef0c1c5c84b9a/internal/rbd/driver/driver.go#L69-L81

Reply:
got it. the aim looks to be new locks instead of sharing as per https://github.com/ceph/ceph-csi/pull/4641#discussion_r1616689529, lets do this one.